aiida/manage/tests/main.py

```diff
@@ -326,9 +325,11 @@ def create_profile(self):
         if not self.root_dir:
             self.root_dir = tempfile.mkdtemp()
         configuration.CONFIG = None
-        settings.AIIDA_CONFIG_FOLDER = self.config_dir
+
+        os.environ[settings.DEFAULT_AIIDA_PATH_VARIABLE] = self.config_dir
```

Comment:
Uh, is it necessary for us to go via modifications of the global environment?
Is there no other way to pass this information along?

I get that this is just for running tests so it's perhaps not as problematic, but in general I think this approach can lead to unexpected results and be hard to debug.
Changing environment variables some way along the way can make

Reply:
I am not sure that there is another way. The problem here is that we need to start the daemon. The location of its log and PID files are in the configuration folder which are indicated by the `AIIDA_PATH` environment variable. When running with a completely temporary test profile, the profile as well as the configuration file and its folder are created from scratch and deleted after the test. The test manager was taking care of resetting the config directory and associated variables in memory of the running test process, however, without these changes, when launching the daemon process, that would still read the original configuration directory (from the `AIIDA_PATH` variable) and so would run in a different directory and nothing would work. I don't see another way currently of doing this then changing the variable and reverting the change at the end of the session. The only other solution would be if we add an option to `verdi` that allows communication of the configuration directory per command. But that is clearly beyond the scope I would say.

Reply:
Ok!
I also saw only later that this code was merely moved here from somewhere else, i.e. indeed not necessary to address here.